blackbox/docs/sql/ddl/basics.txt

```diff
@@ -118,8 +118,52 @@ length. They:
   - should not exceed 255 bytes when encoded with ``utf-8`` (this
     limit applies on the optionally schema-qualified table name)
 
-Column names are restricted in terms of characters. They:
+Column names are restricted in terms of characters.
 
-  - may not contain one of the following characters: ``[`` ``'`` ``]`` ``.``
+  - Columns that *start* with an underscore (``_``) followed by a lowercase
+    letter, followed by a sequence of lower case letters or underscores and
+    ending with a lower case letter are not allowed (Equivalent to
+    ``^_[a-z][_a-z]*[a-z]$``).
+    Such fields are reserved for virtual system columns.
 
-  - may not *start* with an underscore: ``_``
+    Examples:
+
+    +--------------+--------------+
+    | allowed      | not allowed  |
+    +==============+==============+
+    | _foo_        | _foo         |
+    +--------------+--------------+
+    | _Foo         | _foo_bar     |
+    +--------------+--------------+
+    | _fo1o        |              |
+    +--------------+--------------+
+    | __foo        |              |
+    +--------------+--------------+
+
+  - A character sequence ...
+
+    - ... followed by an opening squeare bracket ``[`` ...
+
+    - ... followed by zero or more characters which are not closing square
+      brackets ...
+
+    - ... followed by a closing square braket ``]`` ...
```

Comment:
I think this is a little confusing. I would rephrase it with something like:

```
Character sequences that conform to the subscript notation (for example `abc[abc]`) are
not allowed, as this notation is already used for the identification of columns/paths in
object and array columns.

Additionally, `information_schema.columns` returns columns that are *not* double
quoted. This would lead to conflicts when a table contains object or array columns.
```

Or something like that. Perhaps even a link to the subscript notation at https://crate.io/docs/crate/reference/sql/queries.html#object-arrays would be best as well.